tests/rptest/tests/tiered_storage_model.py

```diff
@@ -0,0 +1,1675 @@
+        if self.__comment is None:
+            short_str = f"{self.__name}:{self.__type}"
+        else:
+            short_str = f"{self.__name}:{self.__type}:{self.__comment}"
+        if len(short_str) < short_len:
+            short_str = short_str.ljust(short_len, '.')
+            return f"Expr[{short_str}]"
+        short_str = short_str[0:(short_len - 1)]
+        short_str = short_str.ljust(short_len, '>')
+        return f"Expr[{short_str}]"
+
+    def __eq__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr == val()
+        return self.__expr == val
+
+    def __lt__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr < val()
+        return self.__expr < val
+
+    def __gt__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr > val()
+        return self.__expr > val
+
+    def __le__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr <= val()
+        return self.__expr <= val
+
+    def __ge__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr >= val()
+        return self.__expr >= val
+
+    def __ne__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr != val()
+        return self.__expr != val
+
+    def __truediv__(self, val):
+        if isinstance(val, Expression):
+            return self.__expr / val()
+        return self.__expr / val
+
+    def make_inputs(self) -> List[TestInput]:
+        """Create set of input for the test"""
+        return []
+
+    def make_validators(self) -> List[EffectValidator]:
+        """Create set of effect observers to run while the data is produced or consumed"""
+        return []
+
+
+class LogBasedValidator(EffectValidator):
+    """Checks that the object is uploaded to S3"""
+    def __init__(self,
+                 name,
+                 pattern,
+                 confidence_threshold=10,
+                 execution_stage=TestRunStage.Produce):
+        self.__confidence_threshold = confidence_threshold
+        self.__name = name
+        self.__num_matches = 0
+        self.__confidence = 0.0
+        self.__pattern = pattern
+        self.__exec_stage = execution_stage
+        self.__logger = None
+
+    def start(self, test: TieredStorageEndToEndTest):
+        """Set initial state using the test instance"""
+        self.__logger = test.get_logger()
+        test.get_logger().info(
+            f"starting validator {self.__name}, pattern {self.__pattern}, threshold {self.__confidence_threshold}"
+        )
+        test.subscribe_to_logs(self, self.__pattern)
+
+    def on_match(self, node_name, line):
+        self.__logger.debug(
+            f"Validator {self.__name} got matching log line {line} on node {node_name}"
+        )
+        self.__num_matches += 1
+        self.__confidence = max(
+            min(self.__num_matches / self.__confidence_threshold, 0.0), 1.0)
```

Comment:
What's the intention here? Looks like it always sets `__confidence` to 1 on match.

Reply:
ditto. for `LogUniquenessValidator.on_match`

Reply:
fixed